src/manifestManager.ts

```diff
@@ -122,8 +122,39 @@ export class ManifestManager implements vscode.Disposable {
         return this.manifests
     }
 
-    getActiveManifest(): Manifest | null {
-        return this.activeManifest
+    /**
+     * Convenience function to get the active manifest and try handle if it doesn't exist.
+     *
+     * This throws an error if a null active manifest is unhandled.
+     *
+     * This also throw error by default when the active manifest contains error.
+     *
+     * @param checkForError Whether to throw an error if any error is found in the manifest
+     * @returns active manifest
+     */
+    async getActiveManifest(checkForError = true): Promise<Manifest> {
```

Comment:
I did a quick review and I am just not happy with the function accepting a param to disable throwing the errors. I understand the use case (for displaying the manifest which might be broken) but i wonder if we could express that differently.

Reply:
An idea: Have a method to get the active manifest uri that doesn't do any checks

Reply:
> An idea: Have a method to get the active manifest uri that doesn't do any checks

Good idea, gonna do that